src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -962,38 +1232,99 @@ function AnalyticsService:GetRemoteConfig(remote: RemoteConfig): { [any]: any }
+				GameAnalytics.HttpApi:initRequest(
+					self._options.gameKey,
+					self._options.secretKey,
+					self._options.build,
+					context.PlayerData,
+					""
+				)
+			else nil
+
 		if server and server.statusCode >= 9 then
 			for _, config in (server.body.configs or {}) do
 				if config.key == remote.name then
 					return resolve(config.value)
 				end
 			end
 		end
-		
+
 		if player and not GameAnalytics:isRemoteConfigsReady(player.UserId) then
 			return resolve(remote.defaultValue)
 		end
-		
-		return resolve(player and GameAnalytics:getRemoteConfigsValueAsString(player.UserId, {
+
+		resolve(player and GameAnalytics:getRemoteConfigsValueAsString(player.UserId, {
 			key = remote.name,
-			defaultValue = remote.defaultValue
+			defaultValue = remote.defaultValue,
 		}) or remote.defaultValue)
 	end)
 end
 
-return AnalyticsService
\ No newline at end of file
+-- GameAnalytics method aliases for custom dimensions
+local dimensionSetter: { [string]: string } = {
+	dimension01 = "setCustomDimension01",
+	dimension02 = "setCustomDimension02",
+	dimension03 = "setCustomDimension03",
+}
+
+--[=[
+	Used to set a custom dimension for a player
+
+	Example Use:
+	```lua
+	AnalyticsService:SetCustomDimension({
+		userId = player.UserId,
+		dimension = "dimension01",
+		value = "value"
+	})
+	```
+
+	To remove a custom dimension from a player, set the value to "".
+
+	For more information about dimensions, refer to [GameAnalytics docs](https://docs.gameanalytics.com/integrations/sdk/roblox/sdk-features?_highlight=dimension#custom-dimensions) on dimensions.
+
+	@param data DimensionData
+	@return Promise
+]=]
+function AnalyticsService:SetCustomDimension(data: DimensionData)
+	return Promise.new(function(resolve, reject)
+		if not self._enabled then
+			return reject("AnalyticsService must be configured with :SetOptions()")
+		elseif not data then
+			return reject("AnalyticsService.SetCustomDimension - data is required")
+		elseif not data.userId then
+			return reject("AnalyticsService.SetCustomDimension - userId is required")
+		elseif not data.dimension then
+			return reject("AnalyticsService.SetCustomDimension - dimension is required")
+		elseif not self._options.customDimensions[data.dimension] then
+			return reject(
+				"AnalyticsService.SetCustomDimension - dimension is invalid, please define it in customDimensions during AnalyticsService:SetOptions()"
+			)
+		elseif not table.find(self._options.customDimensions[data.dimension], data.value) then
+			return reject(
+				"AnalyticsService.SetCustomDimension - dimension value is invalid, please define it as a value in customDimensions during AnalyticsService:SetOptions()"
+			)
+		end
+
+		local setter: string? = dimensionSetter[data.dimension]
+
+		if not setter then
+			return reject("AnalyticsService.SetCustomDimension - dimension is invalid")
+		end
+
+		self:_wrapper(setter, data.userId, data.value)
+
+		resolve()
```

Comment:
```suggestion
		return resolve()
```
The linter/type checker complains if there's a return type defined for a function, but the promise is resolved/rejected without returning the function (For some reason)? It's an invalid error since we know we're returning a promise, but the code checker flags it for some reason.